contrib/babelfishpg_tsql/src/tsqlIface.cpp

```diff
@@ -2681,6 +2681,19 @@ rewriteBatchLevelStatement(
 				bool all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
 				if (all_removed)
 					removeTokenStringFromQuery(expr, cctx->WITH(), ctx);
+
+				/* Check if execute_as_clause is present, and if is present check for caller clause */
+				bool execute_as_clause_check = false;
+				for (auto option : options)
+					if (option->execute_as_clause())
+						execute_as_clause_check = true;
+				if(execute_as_clause_check)
+				{
+					getToken = [](TSqlParser::Function_optionContext* o) { return o->execute_as_clause()->CALLER(); };
+					all_removed = removeTokenFromOptionList(expr, options, commas, ctx, getToken);
+					if (all_removed)
+						removeTokenStringFromQuery(expr, cctx->WITH(), ctx);
+				}
 			}
 		}
 		else if (ctx->create_or_alter_function()->func_body_returns_table()) /* CREATE FUNCTION ... RETURNS TABLE RETURN SELECT ... */
```

Comment:
Why are we not discarding the EXECUTE AS CALLER clause for following conditions?
- CREATE FUNCTION ... RETURNS TABLE RETURN SELECT
- CREATE FUNCTON ... RETURNS INT RETURN
- CREATE OR ALTER PROCEDURE

Reply:
Please check the corresponding tests added for the conditions mentioned above. 
Without making these explicit changes for the conditions, we are getting the desired output [EXECUTE AS CALLER is ignored, the other EXECUTE AS <> are throwing errors].

Reply:
Vasavi mentioned that if we are adding this logic there then we are observing crash that means we are hitting that code that means it is not stale and we should see why above behaviour is observed.